src/opencmiss_iron.f90

```diff
@@ -25199,6 +25201,38 @@ SUBROUTINE cmfe_Equations_JacobianMatricesTypesSet(equations,jacobianTypes,err)
 
   END SUBROUTINE cmfe_Equations_JacobianMatricesTypesSet
 
+  !
+  !================================================================================================================================
+  !
+
+  !>Sets/changes the finite difference step size used for calculating the Jacobian
+  SUBROUTINE cmfe_Equations_JacobianFiniteDifferenceStepSizeSet(equations,jacobianFiniteDifferenceStepSize,err)
```

Comment:
This has a strange interface? It is quite different from the Equations_JacobianGet interface? I would think that to specify a Jacobian to set the finite difference step size it should be the same as getting the Jacobian? There are also multiple step sizes passed via an array?

Reply:
Hi Chris, yup I was very confused about this and why multiple step sizes were needed - I just used a similar function as JacobianMatricesTypesSet - which doesn't work anymore (please see https://github.com/OpenCMISS/iron/issues/160), but will try to use Equations_JacobianGet and hopefully this fixes the issue.

Reply:
Hi @chrispbradley I don't think Equations_JacobianGet will do the job as that is called after Equations are finished, whereas the step size needs to be set during setting up of Equations, Problem is that both cmfe_Equations_JacobianFiniteDifferenceStepSizeSet and JacobianMatricesTypesSet won't work during Equations setup as they require vectorMatrices which I don't think are setup until equations are finished (#160 ) - most likely need to be cached. Unfortunately, Mario can't move forward without these and it has already taken too long to try and set a simple flag. I am going to place the stepsize and Jacobian type set directly in the equations type and get it going so that he can move forward with debugging the active contraction (still doesn't work). Then can get back to fixing this setup issue. Thanks

Reply:
Once fixed, will need to add some more cases to the finite elasticity tests to make sure future code changes that effect it are flagged.

Reply:
Added more info to https://github.com/OpenCMISS/iron/issues/160 regarding the source of the issue

Reply:
Hi Prasad, sure, add it to the equations. Maybe make the routine name Equations_FiniteDifferenceStepSizeSet since this will apply to the equations rather than just Jacobians? I'm also not sure why JacobianVariableTypesSet is a public routine???

Reply:
Thanks, will do. The JacobianVariableTypesSet was added by Jenny to allow a switch between calculating analytic and FD Jacobians inside OpenCMISS. These options only work if nonlinearSolver.NewtonJacobianCalculationTypeSet(
    iron.JacobianCalculationTypes.EQUATIONS) 
If nonlinearSolver.NewtonJacobianCalculationTypeSet(
    iron.JacobianCalculationTypes.FD) then the petsc's FD calculates the Jacobian

Reply:
Hmm, but the variableType for the Jacobian is the field variable the residual is differentiated by to give the Jacobian. This is something that is defined by the variable types for the equations. I'm not sure I get why a user would want or need to change them?

Reply:
yup, thats a bit weird, we probably need to update the name to equations.JacobianCalculationType?